### DISCUSSION TITLE
Add a "search for x in directory"

### DISCUSSION THREAD

**Discussion by Milo123459** [community/community#9405] [Other Feature Feedback, Questions, & Ideas] [answered]
<img width="284" alt="image" src="https://user-images.githubusercontent.com/50248166/147714477-c030d23a-8bdc-435f-bfdf-bf08a88ba971.png">

It would be nice if there was a button to search for code in an inputted folder, for example:
<img width="960" alt="ysTq8002" src="https://user-images.githubusercontent.com/50248166/147714523-80457170-68b8-46f0-a1e8-f7ad2b3f1993.png">

As seen, I am searching in `extensions/typescript-language-features/src` - there could be a button that searches for that inside of that directory I have inputted.

**Comment by look** [comment, marked as answer]
Thanks for the feedback! Searching for code in a directory is possible using the `path:` qualifier, for example:

```
repo:microsoft/vscode path:/^extensions\/typescript-language-features\/src\/tsServer/ _validateTypeScript
```

However, there is no UX for that when viewing a file (you will see a search box while viewing a directory, though). I'll raise your feedback to the team.

**Reply by Milo123459** [reply]
Thank you! Please keep me posted on anything new, and if this gets implemented!
